File: 0078-subsets/0078-subsets.cpp
class Solution {
public:
    vector<vector<int>> res;
    vector<int> list;
    vector<vector<int>> subsets(vector<int>& nums) {
        backtracking(nums, 0);
        return res;
    }

    void backtracking(vector<int>& nums, int start) {
        vector<int> copy = list;
        res.push_back(copy);

        if (start == nums.size()) {
            return;
        }

        for (int i = start; i < nums.size(); i++) {
            list.push_back(nums[i]);
            backtracking(nums, i + 1);
            list.pop_back();
        }
    }
};

File: 0078-subsets/0078-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        path = []
        res = []
    
        def backtracking(start: int):
            res.append(path.copy())
            
            if start == len(nums):
                return
            
            for i in range(start, len(nums)):
                path.append(nums[i])
                backtracking(i + 1)
                path.pop()
        
        backtracking(0)
        
        return res